packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -59,6 +70,49 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
     return documentMetadataService.getStatus(document, meta.availableStatus);
   };
 
+  /**
+   * Get the difference between the version schema and the content type schema
+   * Returns the attributes with their original shape
+   */
+  const getAttributesDiff = (
```

Comment:
fyi, there is a utility to diff schemas in `packages/core/database/src/schema/diff.ts` , it doesn't look like it's exported though. I have still not looked at the details but it's kind of accessible through `strapi.db.schema.schemaDiff.diff()` , if you are interested we could sync with alex to check if it's ok to use that here

Reply:
code is not long, so I don't think its a bad idea to keep your implementation either. Mentioned just in case!

Reply:
🤦‍♂️ didn't even think to look if something already existed. I'll take a look thanks.

Reply:
I took a look but it does look like it is diffing database columns so I don't think it is applicable here.

Reply:
🫡 makes sense